build-system/tasks/remap-dependencies-plugin/test-remap-dependencies.js

```diff
@@ -0,0 +1,216 @@
+ */
+const resolveMock = sinon.fake((s) => {
+  if (s.startsWith('node-mod-')) {
+    // return input if it's a node module (use special "node-mod-" prefix in tests to indicate node module)
+    return s;
+  } else {
+    if (!s.startsWith('.')) {
+      // otherwise, assume local module,
+      // ensure local modules always have leading .
+      s = `./${s}`;
+    }
+    // handle directory imports with barrel file and appropriate file extension to emulate behavior of resolvePath
+    if (s.match(/((js|jsx|ts|tsx)-)?dir\d?(\/index)?$/)) {
+      // return barrel file if importing a directory (use special "dir" in tests to indicate path to directory)
+      const [, , fileType, indexStr] = s.match(
+        /((js|jsx|ts|tsx)-)?dir\d?(\/index)?$/
+      );
+      s = `${s}${!!indexStr ? '' : '/index'}.${fileType || 'js'}`;
+    }
+    // handle file imports with appropriate file extension to emulate behavior of resolvePath
+    if (s.match(/\/((js|jsx|ts|tsx)-)?mod\d?$/)) {
+      const fileType = s.match(/\/((js|jsx|ts|tsx)-)?mod\d?$/)[2];
+      s = `${s}.${fileType || 'js'}`;
+    }
+    return s;
+  }
+});
+
+/**
+ * @param {Array<string>} externals
+ * @param {Object} remaps
+ * @return {*}
+ */
+function setup(externals, remaps) {
+  // indicates that a particular path was resolved relative to the root of the directory
+  // const resolvePathSpy = sinon.fake((s) => (resolve ? resolve(s) : `${s}`));
+  const plugin = remapDependenciesPlugin({
+    externals,
+    remaps,
+    resolve: resolveMock,
+  });
+  const onResolveSpy = sinon.spy();
+  plugin.setup({onResolve: onResolveSpy});
+  const onResolve = onResolveSpy.getCall(0).args[1];
+  return onResolve;
+}
+
+test('remap node modules to other node modules', (t) => {
+  const onResolve = setup(['node-mod-A', 'node-mod-C'], {
+    // multiple mods to same external mod
+    'node-mod-1': 'node-mod-A',
+    'node-mod-2': 'node-mod-A',
+
+    // multiple mods to same bundled mod
+    'node-mod-3': 'node-mod-B',
+    'node-mod-4': 'node-mod-B',
+
+    // single mod to external mod
+    'node-mod-5': 'node-mod-C',
+
+    // single mod to bundled mod
+    'node-mod-6': 'node-mod-D',
+  });
+
+  const res1 = onResolve({path: 'node-mod-1'});
+  t.is(res1?.path, 'node-mod-A');
+  t.is(res1?.external, true);
+
+  const res2 = onResolve({path: 'node-mod-2'});
+  t.is(res2?.path, 'node-mod-A');
+  t.is(res2?.external, true);
+
+  const res3 = onResolve({path: 'node-mod-3'});
+  t.is(res3?.path, 'node-mod-B');
+  t.is(res3?.external, false);
+
+  const res4 = onResolve({path: 'node-mod-4'});
+  t.is(res4?.path, 'node-mod-B');
+  t.is(res4?.external, false);
+
+  const res5 = onResolve({path: 'node-mod-5'});
+  t.is(res5?.path, 'node-mod-C');
+  t.is(res5?.external, true);
+
+  const res6 = onResolve({path: 'node-mod-6'});
+  t.is(res6?.path, 'node-mod-D');
+  t.is(res6?.external, false);
+
+  // remapping module with no remaps should return undefined
+  t.is(onResolve({path: 'mod0'}), undefined);
```

Comment:
nit: shouldn't this be in its own test?

Reply:
this was actually supposed to test importing a node module that didn't have remaps, so I'll actually change the path from `'mod0'` to `'node-mod-0'` to align with the other paths